cloudlift/deployment/service_template_generator.py

```diff
@@ -700,19 +674,273 @@ def _add_alb(self, cd, service_name, config, launch_type):
+
+
+    def _create_listener_rules(
+            self, service_name: str, is_internal: bool, hostnames: list[str], alb_listeners: list[dict[str, str]], target_group_arn: str
+    ) -> list[ListenerRule]:
+        if not alb_listeners:
+            raise UnrecoverableException("No listener found on the cluster ALB")
+
+        http_listener_arns: list[str] = []
+        https_listener_arns: list[str] = []
+
+        for listener in alb_listeners:
+            key = listener.get("Key")
+            if "HTTP" in key and "HTTPS" not in key:
+                if is_internal:
+                    # for internet facing services, we only need to create listener rules for HTTPS listener
+                    http_listener_arns.append(listener.get("Value"))
+            elif "HTTPS" in key:
+                https_listener_arns.append(listener.get("Value"))
+
+        # listener rules
+        http_listener_rules: list[ListenerRule] = []
+        https_listener_rules: list[ListenerRule] = []
+
+        # create listener rules for HTTP and HTTPS listeners
+        for index, http_listener_arn in enumerate(http_listener_arns):
+            index = index + 1
+
+            priority = self._get_listener_rule_priority(http_listener_arn, hostnames[0])
+            http_listener_rule = self._get_listener_rule("HTTP", index, is_internal, http_listener_arn, priority, hostnames, service_name, target_group_arn)
+
+            http_listener_rules.append(http_listener_rule)
+
+        for index, https_listener_arn in enumerate(https_listener_arns):
+            index = index + 1
+
+            priority = self._get_listener_rule_priority(https_listener_arn, hostnames[0])
+            https_listener_rule = self._get_listener_rule("HTTPS", index, is_internal, https_listener_arn, priority, hostnames, service_name, target_group_arn)
+
+            https_listener_rules.append(https_listener_rule)
+
+        if is_internal:
+            for rule in http_listener_rules:
+                self.template.add_resource(rule)
+            for rule in https_listener_rules:
+                self.template.add_resource(rule)
+        else:
+            for rule in https_listener_rules:
+                self.template.add_resource(rule)
+        return https_listener_rules + http_listener_rules
+
+    def _get_listener_rule_priority(self, listener_arn: str, hostname: str) -> int:
+        MIN_PRIORITY = 100
+        MAX_PRIORITY = 49999
+
+        # get all the listener rules in the listener
+        elbv2_client = get_client_for('elbv2', self.env)
+        rules = elbv2_client.describe_rules(ListenerArn=listener_arn, PageSize=400)
+
+        # check if there's already a rule with same host-header condition
+        for rule in rules['Rules']:
+            for condition in rule['Conditions']:
+                if condition['Field'] == 'host-header' and hostname in condition['HostHeaderConfig']['Values']:
+                    return int(rule['Priority'])
```

Comment:
Need to also check that the priority of the rule returned is greater than `MIN_PRIORITY`

Reply:
changed the behavior, reduced MAX_PRIORITY from  49999 to 49899, room for 100 more custom rules after the priorities generated by the code.